Base/home/anon/tests/run-tests-and-shutdown.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+echo "==== Running Tests on SerenityOS ===="
+
+run() {
+    test_cmd=($*)
+    echo "Running test $idx/$count -- $test_cmd"
+    if $test_cmd {
+        echo "::info file=$test_cmd:: $test_cmd passed!"
+    } else {
+        echo "::error file=$test_cmd:: $test_cmd returned non-zero exit code, check logs!"
+    }
+    idx=$(expr $idx + 1)
+}
+
+# TODO: It'd be nice to have a list+list op (as opposed to nest-on-in-another)
+system_tests=(test-js (test-crypto bigint -t)) #("test-js" "test-web" "test-pthread" "test-crypto" "test-compress" $usr_tests_tests)
+all_tests=($system_tests $(find /usr/Tests -type f | grep -v Kernel | grep -v .inc | shuf))
```

Comment:
Shuffling them will reveal more bugs, at the cost of being non-deterministic.

I agree with you that we should reveal more bugs, but in order to track them down, the order needs to be recoverable. Maybe print the names of the tests as they are being executed, otherwise we will end up with randomly failing tests.

Reply:
The names _are_ being printed, take a look at the `run()` function.

Reply:
D'oh! Yes, you're right.